src/main/java/org/myrobotlab/service/MqttBroker.java

```diff
@@ -0,0 +1,628 @@
+import java.security.MessageDigest;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.MRLListener;
+import org.myrobotlab.framework.Message;
+import org.myrobotlab.framework.MethodCache;
+import org.myrobotlab.framework.Service;
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.logging.LoggingFactory;
+import org.myrobotlab.mqtt.MqttMsg;
+import org.myrobotlab.net.Connection;
+import org.myrobotlab.service.interfaces.Gateway;
+import org.slf4j.Logger;
+
+import io.moquette.broker.Server;
+import io.moquette.broker.config.MemoryConfig;
+import io.moquette.interception.InterceptHandler;
+import io.moquette.interception.messages.InterceptAcknowledgedMessage;
+import io.moquette.interception.messages.InterceptConnectMessage;
+import io.moquette.interception.messages.InterceptConnectionLostMessage;
+import io.moquette.interception.messages.InterceptDisconnectMessage;
+import io.moquette.interception.messages.InterceptMessage;
+import io.moquette.interception.messages.InterceptPublishMessage;
+import io.moquette.interception.messages.InterceptSubscribeMessage;
+import io.moquette.interception.messages.InterceptUnsubscribeMessage;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.mqtt.MqttMessageBuilders;
+import io.netty.handler.codec.mqtt.MqttPublishMessage;
+import io.netty.handler.codec.mqtt.MqttQoS;
+
+/**
+ * Embedded moquette mqtt broker. It has the same capability of a standard mqtt
+ * broker in addition it has an interface to the mrl pub/sub framework.
+ * 
+ * The mrl framework is accessible through a topic with a prefix. Default prefix
+ * is "mrl". The topic at that point follows /{serviceName}/{method} with the
+ * payload being a json array with encoded parameters e.g.
+ * /mrl/clock01/setInterval with payload ['2000']
+ * 
+ * this will invoke clock01.setInterval(Integer ms) Make note: the parameter
+ * list values are encoded, that is why its '2000' vs 2000, because first the
+ * whole array is decoded - then each parameter is, similar to http form values
+ *
+ */
+public class MqttBroker extends Service implements InterceptHandler, Gateway {
+
+  public final static Logger log = LoggerFactory.getLogger(MqttBroker.class);
+
+  private static final long serialVersionUID = 1L;
+
+  private static String bytesToHex(byte[] hash) {
+    StringBuilder hexString = new StringBuilder(2 * hash.length);
+    for (int i = 0; i < hash.length; i++) {
+      String hex = Integer.toHexString(0xff & hash[i]);
+      if (hex.length() == 1) {
+        hexString.append('0');
+      }
+      hexString.append(hex);
+    }
+    return hexString.toString();
+  }
+
+  private static final MqttMsg toMqttMsg(InterceptMessage msg) {

```

Comment:
why would you want it to be both private and static?  only used by the current class in a static context?

Reply:
was hoping mqtt client and mqtt broker could share the logic of converting a mqtt msg into a pojo ...
they can't - it will be private